File: engine/model/src/main/scala/tools/packet/message/GiveFameResponse.scala
package tools.packet.message

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GiveFameResponse(private var _mode: Int, private var _characterName: String, private var _newFame: Int) extends PacketInput {
  def mode: Int = _mode

  def characterName: String = _characterName

  def newFame: Int = _newFame

  override def opcode(): SendOpcode = SendOpcode.FAME_RESPONSE
}

File: engine/engine-model/src/main/java/tools/packet/message/GiveFameResponse.java
package tools.packet.message;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GiveFameResponse(Integer mode, String characterName, Integer newFame) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.FAME_RESPONSE;
   }
}